### PR TITLE
Adds __repr__ method to Token - Fixes issue 7250

### PR DESCRIPTION
**Pull request by ErickGiffoni**
According to the previous conversation in #9543 , I followed the instructions and created this new PR.

**Previous notes**

Issue #7250 was previously solved via PR #7839 but there was some trouble during the checks.
The conversation in the PR 7839 ended, so I continued the work.

**Proposed changes**:
- Represent a `Token` object as a string using python `__repr__` method
- Add a change log related to that

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

Hey @aeshky, this is the new PR you asked me to create and tag you :) .

### CODE REVIEW

**Comment by ErickGiffoni** [comment]
I see the PR failed the code quality check.
I'm on it to fix it !

**Comment by aeshky** [comment]
Hi @ErickGiffoni. Yes, it's saying the line with return statement is too long. You should split it then reformat using `black`.

**Comment by ErickGiffoni** [comment]
@aeshky thanks, I've done it. Now it should be ok !

**Comment by ErickGiffoni** [comment]
Ok, I'm on it again lol

**Comment by ErickGiffoni** [comment]
@aeshky what do you think we should do about the missing docstring problem in the Code Quality check ?

Can I move the docstring under the `Tokenizer.__init__` up to line 70 ?

File: rasa/nlu/tokenizers/tokenizer.py [hunk 1]
```diff
@@ -61,6 +61,10 @@ def __lt__(self, other: Any) -> bool:
             other.lemma,
         )
 
+    def __repr__(self) -> Text:
+        return f"<Token object value='{self.text}' start={self.start} end={self.end} \
+        at {hex(id(self))}>"
+
 
 class Tokenizer(Component):
```
Comment (aeshky) [inline rasa/nlu/tokenizers/tokenizer.py:69]: """Base class for tokenizers."""

Comment (aeshky) [inline rasa/nlu/tokenizers/tokenizer.py:69]:
```suggestion
class Tokenizer(Component):
    """Base class for tokenizers."""

```

Comment (aeshky) [reply rasa/nlu/tokenizers/tokenizer.py:69]: @ErickGiffoni please accept this change to resolve the code quality error. 
I ran the checks locally and the linter will also ask you to remove the empty lines after the docstrings of the class methods. 
These changes should allow the tests to pass (hopefully) 👍

Comment (aeshky) [reply rasa/nlu/tokenizers/tokenizer.py:69]: If you want to check code quality locally before committing you can run the following commands:
```
make formatter
make types
make lint
```

Comment (ErickGiffoni) [reply rasa/nlu/tokenizers/tokenizer.py:69]: Nice ! Thanks !

**Comment by ErickGiffoni** [comment]
@aeshky are we the champions ? 😆 
I think now we can probably celebrate

**Review by aeshky — CHANGES_REQUESTED** [review]
Looks great @ErickGiffoni and all the tests are passing 🚀
Please accept my edit to the changelog and update the branch, and I'll approve your request.

**Comment by aeshky** [comment]
Hi @ErickGiffoni, I'm wondering how you're getting on with this issue. Do you need any further support? Only a small changelog edit and a branch update left before we can merge this 🙂

**Comment by ErickGiffoni** [comment]
> Hi @ErickGiffoni, I'm wondering how you're getting on with this issue. Do you need any further support? Only a small changelog edit and a branch update left before we can merge this 🙂

Hey @aeshky sorry for the delay ! I was facing some personal problems.
I 'll get to it right now ! Thanks !

**Review by aeshky — APPROVED** [review]

**Comment by aeshky** [comment]
Congrats on merging this @ErickGiffoni and thanks for your contribution 🚀

**Pull request merged**